lib/strategies/backoff_strategy.rb

```diff
@@ -0,0 +1,71 @@
+module Strategies
+  class BackoffStrategy
+    PROTECT_DURATION = 60 * 60 * 24 # eq 24 hours in seconds
+    JITTER_VARIANCE = 0.2
+
+    def initialize(key:, storage:, config:)
+      @storage = storage
+      @config = default_config.merge(config)
+      @id = "fail_to_ban:#{key}"
+    end
+
+    def call
+      true
+    end
+
+    def attempt
+      return false if blocked?
+      increment_failed_attempts
+      true
+    end
+
+    def blocked?
+      return false if retry_count < @config[:permitted_attempts] || Time.now.to_i >= unlock_at
```

Comment:
when returning false if a condition and then true makes me feel we should use the condition itself as the return value